### ISSUE TITLE
PDF renders incorrectly if wrapped in an element with dir="rtl"

### ISSUE THREAD

**Issue by BoazKG93** [wojtekmaj/react-pdf#588]
So I'm using this module in a fairly simple way as follows (my build is standard react-create-app):

```jsx
import { Document, Page } from "react-pdf";
import { pdfjs } from "react-pdf";
pdfjs.GlobalWorkerOptions.workerSrc = `//cdnjs.cloudflare.com/ajax/libs/pdf.js/${pdfjs.version}/pdf.worker.js`;

<Document
  file={`${this.state.base64Url}`}
  options={{
    cMapUrl: `//cdn.jsdelivr.net/npm/pdfjs-dist@${pdfjs.version}/cmaps/`,
    cMapPacked: true
  }}
  onLoadError={(err: any) => {
    console.log(err);
  }}
>
  <Page pageNumber={1} />
</Document>
```

The PDF renders the text very weirdly like that:

![](https://i.imgur.com/aLWGHFZ.png)

I'm using the simple dummy pdf from: https://www.w3.org/WAI/ER/tests/xhtml/testfiles/resources/pdf/dummy.pdf

I can't figure out what's the problem. My base64 is ok and I doubled checked it online to see it renders correctly. No matter which PDF I try, I always get this.

I don't have any errors on my console, other than this warning:
```
./node_modules/pdfjs-dist/build/pdf.js
Critical dependency: require function is used in a way in which dependencies cannot be statically extracted
```

**Comment by malachitetrip** [comment]
Same issue

**Comment by zachi** [comment]
i have the same issue, @malachitetrip removing global styles didnt help.  
would love some guidance

**Comment by wojtekmaj** [comment]
Could not reproduce when uploading this to my [test suite](https://projects.wojtekmaj.pl/react-pdf/test/), neither on Firefox nor Chrome. With what browser and React-PDF version are you able to reproduce that with?

**Comment by zachi** [comment]
im using '5.0.0-beta.4' version (in order to avoid a different issue), and it happens in all browsers.

**Comment by wojtekmaj** [comment]
Doesn't seem to be the case... I tried reproducing this error, both on 4.1.0 stable and 5.0.0-beta.4 Dummy PDF file displays absolutely fine on Chrome and Firefox.

https://codesandbox.io/s/react-pdf-dummy-pdf-9q27q

Does anyone see the above link with PDF wrongly rendered? If so, please share your setup with me so I could reproduce and see where the issue is.

**Comment by zachi** [comment]
ok found it, seems like <body dir="rtl" is causing the funky render. you can see the problem here:
[https://codesandbox.io/s/react-pdf-dummy-pdf-zwo0h?file=/index.html](url)
fixed it on my end, by removing 'dir="rtl"' from the the document viewer (not necessary). @wojtekmaj i might consider fixing the package

**Comment by wojtekmaj** [comment]
👏  👏  👏  Amazing job!

"dir" can actually be set on per-element level, so it would be as easy as wrapping the document in `<div dir="ltr" />`, without changing the `dir` attribute on `body`.

Seems like this needs to be fixed on PDF.js end, though. I'll report it.

**Comment by wojtekmaj** [comment]
Note: This fix was cherry-picked to [v4.2.0](https://github.com/wojtekmaj/react-pdf/releases/tag/v4.2.0) release.
